pkgs/development/compilers/openjdk/jre.nix

```diff
@@ -0,0 +1,19 @@
+{ jdk
+, runCommand
+, patchelf
+, lib
+, modules ? [ "java.base" ]
+}:
+
+let
+  jre = runCommand "${jdk.name}-jre" rec {
+    nativeBuildInputs = [ patchelf ];
+    buildInputs = [ jdk ];
+    passthru = {
+      home = "${jre}";
+    };
+  }   ''
+      jlink --module-path ${jdk}/lib/openjdk/jmods --add-modules ${lib.concatStringsSep "," modules} --output $out
+      patchelf --shrink-rpath $out/bin/* $out/lib/jexec $out/lib/jspawnhelper $out/lib/*.so $out/lib/*/*.so
+  '';
+in jre
```

Comment:
why have this `let ... in jre` structure and not just `runCommand`?

Reply:
AFAICS this is needed to recursively refer to `jre` on line 13 - or is there another way to refer to 'yourself'?